frameworks/helloworld/src/main/java/com/mesosphere/sdk/helloworld/scheduler/Main.java

```diff
@@ -18,35 +21,51 @@
     private static final String TASK_NAME = "hello";
 
     public static void main(String[] args) throws Exception {
-        SchedulerConfig schedulerConfig = SchedulerConfig.fromEnv();
-        if (args.length > 0) {
-            new DefaultService(new File(args[0]), schedulerConfig).run();
-        } else {
-            // Example of building a custom ServiceSpec entirely in Java without a YAML file:
-            TaskEnvRouter taskEnvRouter = new TaskEnvRouter();
-            new DefaultService(DefaultServiceSpec.newBuilder()
-                    .name("hello-world")
-                    .principal("hello-world-principal")
-                    .zookeeperConnection("master.mesos:2181")
-                    .addPod(DefaultPodSpec.newBuilder(schedulerConfig.getExecutorURI())
-                            .count(COUNT)
-                            .type(POD_TYPE)
-                            .addTask(DefaultTaskSpec.newBuilder()
-                                    .name(TASK_NAME)
-                                    .goalState(GoalState.RUNNING)
-                                    .commandSpec(DefaultCommandSpec.newBuilder(taskEnvRouter.getConfig(POD_TYPE))
-                                            .value("echo hello >> hello-container-path/output && sleep 1000")
-                                            .build())
-                                    .resourceSet(DefaultResourceSet
-                                            .newBuilder("hello-world-role", Constants.ANY_ROLE, "hello-world-principal")
-                                            .id("hello-resources")
-                                            .cpus(CPUS)
-                                            .memory(256.0)
-                                            .addVolume("ROOT", 5000.0, "hello-container-path")
-                                            .build()).build()).build()).build(),
+        final SchedulerConfig schedulerConfig = SchedulerConfig.fromEnv();
+        final SchedulerRunner runner;
+        if (args.length == 0) {
+            // Create a sample config in Java
+            runner = SchedulerRunner.fromServiceSpec(
+                    createSampleServiceSpec(schedulerConfig), schedulerConfig, Collections.emptyList());
+        } else if (args.length == 1) {
+            // Read config from provided file, assume any config templates are in the same directory as the file
```

Comment:
There's still some minor differences between how this is handled, and Template for example.

Reply:
Updated the two to be more similar. The only difference now is hello-world's "here's a ServiceSpec from scratch" example.